cobalt/browser/user_agent/user_agent_platform_info.cc

```diff
@@ -205,15 +206,14 @@ void InitializeUserAgentPlatformInfoFields(UserAgentPlatformInfo& info) {
   }
 #endif  // ENABLE_DEBUG_COMMAND_LINE_SWITCHES
 
-#if BUILDFLAG(IS_ANDROID)
-  info.set_device_type("ATV");
-#else
-  info.set_device_type("TV");
-#endif  // BUILDFLAG(IS_ANDROID)
+  info.set_device_type(
+      starboard::GetSystemPropertyString(kSbSystemPropertyDeviceType));
 
-// TODO(cobalt, b/374213479): figure out firmware version for other platforms.
 #if BUILDFLAG(IS_ANDROID)
   info.set_firmware_version(base::SysInfo::GetAndroidBuildID());
+#elif BUILDFLAG(IS_STARBOARD)
+  info.set_firmware_version(
+      starboard::GetSystemPropertyString(kSbSystemPropertyFirmwareVersion));
 #endif  // BUILDFLAG(IS_ANDROID)
 
   info.set_model(base::SysInfo::HardwareModelName());
```

Comment:
Instead of adding more code to base/sys_info_starboard.cc,
I'd split here
```
#if BUILDFLAG(IS_STARBOARD)
  info.set_model(GetSystemPropertyString(kSbSystemPropertyModelName));
#else.
  info.set_model(base::SysInfo::HardwareModelName());
#endif
```

This is because ideally we'd avoid modifying //base (even for 
net additions), specially for a use case like this, where 
//cobalt can call directly to //starboard. It'd also simplify
my rebasing of #7076 :)